smoke/IotEdgeQuickstart/details/IotedgedWindows.cs

```diff
@@ -0,0 +1,147 @@
+namespace IotEdgeQuickstart.Details
+{
+    using System;
+    using System.ComponentModel;
+    using System.IO;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    class IotedgedWindows : IBootstrapper
+    {
+        readonly string archivePath;
+        readonly Option<RegistryCredentials> credentials;
+        string scriptDir;
+
+        public IotedgedWindows(string archivePath, Option<RegistryCredentials> credentials)
+        {
+            this.archivePath = archivePath;
+            this.credentials = credentials;
+        }
+
+        public async Task VerifyNotActive()
+        {
+            bool active = true;
+            try
+            {
+                await Process.RunAsync("sc", "query iotedge");
```

Comment:
Is this going to run on the host on Windows? If yes, we may be able to use the [service controller](https://docs.microsoft.com/en-us/dotnet/api/system.serviceprocess.servicecontroller?view=netframework-4.7.2) API for this. Its a .NET Framework only thing though.

Reply:
Well yes this would run on the windows host, but we're compiling for .NET Core, so...

Reply:
...although the doc you linked to said it's supported on .NET Core as well.

Reply:
This bit of code has been hassle-free so far so I'll leave it. But we can change it whenever it suites us.

Reply:
Sounds good